docs/query-dsl/bool-dsl/bool-dsl.asciidoc

```diff
@@ -364,3 +364,55 @@ Assert(
     c => AssertDoesNotJoinOntoLockedBool(c, "leftBool"));
 ----
 
+=== Perfomance considerations
+
+If you have a requirement of combining many many queries using the bool dsl please take the following into account.
+
+You *can* use bitwise assignments in a loop to combine many queries into a bigger bool.
+
+NOTE: This needs NEST 2.4.6 or higher. Prior versions did not do a good job flattening the result in the most optimal way and could cause a stackoverflow when doing ~2000 iterations In later version you can combine as many as you'd like but please keep mind of the following performance recommendations
+

```

Comment:
This needs to be updated to specifically mention we did not do a good job combining many bools this way, other queries were flattened correctly.